File: frontend/src/types/job.ts
/** 岗位类型定义 */

export interface Job {
  id: string;
  name: string;
  category: 'hr' | 'software' | 'finance' | 'sales';
  description?: string;
  required_skills: string[];
  preferred_skills: string[];
  min_work_years: number;
  min_education: string;
  skill_weight: number;
  experience_weight: number;
  education_weight: number;
  pass_threshold: number;
  review_threshold: number;
  is_active: boolean;
}

export interface JobCreate {
  name: string;
  category: 'hr' | 'software' | 'finance' | 'sales';
  description?: string;
  required_skills: string[];
  preferred_skills: string[];
  min_work_years: number;
  min_education: string;
  skill_weight?: number;
  experience_weight?: number;
  education_weight?: number;
  pass_threshold?: number;
  review_threshold?: number;
}

export interface JobUpdate {
  name?: string;
  category?: 'hr' | 'software' | 'finance' | 'sales';
  description?: string;
  required_skills?: string[];
  preferred_skills?: string[];
  min_work_years?: number;
  min_education?: string;
  skill_weight?: number;
  experience_weight?: number;
  education_weight?: number;
  pass_threshold?: number;
  review_threshold?: number;
  is_active?: boolean;
}

// 岗位类别显示名称映射
export const JOB_CATEGORY_LABELS: Record<Job['category'], string> = {
  hr: 'HR岗位',
  software: '软件开发',
  finance: '财务岗位',
  sales: '销售岗位'
};

// 岗位类别颜色映射
export const JOB_CATEGORY_COLORS: Record<Job['category'], string> = {
  hr: 'purple',
  software: 'blue',
  finance: 'green',
  sales: 'orange'
};

File: backend/app/schemas/job.py
"""岗位相关的Pydantic schemas"""
from pydantic import BaseModel, Field
from typing import List, Optional
from uuid import UUID


class JobBase(BaseModel):
    """岗位基础模型"""
    name: str = Field(..., description="岗位名称")
    category: str = Field(..., description="岗位类别: hr/software/finance/sales")
    description: Optional[str] = Field(None, description="岗位描述")
    required_skills: List[str] = Field(default_factory=list, description="必备技能")
    preferred_skills: List[str] = Field(default_factory=list, description="加分技能")
    min_work_years: int = Field(0, description="最低工作年限")
    min_education: str = Field("大专", description="最低学历")
    skill_weight: int = Field(50, description="技能权重")
    experience_weight: int = Field(30, description="经验权重")
    education_weight: int = Field(20, description="学历权重")
    pass_threshold: int = Field(70, description="PASS阈值")
    review_threshold: int = Field(50, description="REVIEW阈值")


class JobCreate(JobBase):
    """创建岗位"""
    pass


class JobUpdate(BaseModel):
    """更新岗位"""
    name: Optional[str] = None
    category: Optional[str] = None
    description: Optional[str] = None
    required_skills: Optional[List[str]] = None
    preferred_skills: Optional[List[str]] = None
    min_work_years: Optional[int] = None
    min_education: Optional[str] = None
    skill_weight: Optional[int] = None
    experience_weight: Optional[int] = None
    education_weight: Optional[int] = None
    pass_threshold: Optional[int] = None
    review_threshold: Optional[int] = None
    is_active: Optional[bool] = None


class JobResponse(JobBase):
    """岗位响应"""
    id: UUID
    is_active: bool

    class Config:
        from_attributes = True
